R/scatter_plot_utils.R

```diff
@@ -280,8 +341,25 @@ static_proj_data <- function(plot_model, df_point, df_ribbon,
       }
     }
   }
-
+  if (!is.null(df_sample)) {
```

Comment:
Same here, really confusing code that could benefit from modularisation and more comments